language-plutus-core/prelude/PlutusPrelude.hs

```diff
@@ -211,6 +231,27 @@ instance Functor f => Functor (PairT b f) where
 hoist :: (Recursive t, Corecursive t) => (Base t t -> Base t t) -> t -> t
 hoist f = c where c = embed . f . fmap c . project
 
+type family UnwrapRep r :: * where
+    UnwrapRep (D1 d (C1 c (S1 s (K1 i a)))) = a
+
+-- | Extract the underlying type of a newtype wrapper.
+type family Unwrap a :: * where
```

Comment:
Methinks you want http://hackage.haskell.org/package/lens-4.17/docs/Control-Lens-Wrapped.html, but I guess we're not brave enough for a `lens` depdendency yet.

Reply:
I would use that if we had the `lens` dependency, indeed, but the current solution is rather light-weight, so `lens` wouldn't buy us much here.